app/plugin.go

```diff
@@ -257,9 +257,7 @@ func (a *App) SyncPlugins() *model.AppError {
 }
 
 func (a *App) ShutDownPlugins() {
```

Comment:
Another question for my understanding: Where does the lock get acquired, when the plugin calls `CreatePost`? I can find the code piece for this.

Reply:
`CreatePost` attempts to acquire it but is blocked because `ShutDownPlugins` already has read/write lock acquired.

`CreatePost` attempts to acquire it [here](https://github.com/mattermost/mattermost-server/blob/master/app/post.go#L240)